src/d3a/models/strategy/external_strategies/__init__.py

```diff
@@ -39,227 +46,286 @@
 
-    def _remove_area_uuid_from_aggregator_mapping(self):
-        self.redis.aggregator.device_aggregator_mapping.pop(self.device.uuid, None)
-
     @property
-    def should_use_default_strategy(self):
-        return self._use_template_strategy or \
-               not (self.connected or self.is_aggregator_controlled)
+    def should_use_default_strategy(self) -> bool:
+        """
+        Definite (decisive) flag to ignore external/ overridden behaviors and
+        resort to the template's default strategy.
+        """
+        return (self._use_template_strategy or
+                not (self.connected or self.is_aggregator_controlled))
 
     @staticmethod
-    def _get_transaction_id(payload):
+    def _get_transaction_id(payload: Dict) -> str:
+        """Extract the transaction_id from the received payload."""
         data = json.loads(payload["data"])
-        if key_in_dict_and_not_none(data, "transaction_id"):
+        if data.get("transaction_id"):
             return data["transaction_id"]
-        else:
-            raise ValueError("transaction_id not in payload or None")
+        raise ValueError("transaction_id not in payload or None")
 
-    def area_reconfigure_event(self, *args, **kwargs):
+    def area_reconfigure_event(self, *args, **kwargs) -> None:
         """Reconfigure the device properties at runtime using the provided arguments."""
-        if key_in_dict_and_not_none(kwargs, "allow_external_connection"):
+        if kwargs.get("allow_external_connection"):
             self._use_template_strategy = not kwargs["allow_external_connection"]
         super().area_reconfigure_event(*args, **kwargs)
 
-    def _register(self, payload):
-        self._connected = register_area(self.redis, self.channel_prefix, self.connected,
-                                        self._get_transaction_id(payload),
-                                        area_uuid=self.device.uuid)
+    def _register(self, payload: Dict) -> None:
+        """Callback for the register redis command."""
+        self._is_registered = ExternalStrategyConnectionManager.register(
+            self.redis, self.channel_prefix, self.connected,
+            self._get_transaction_id(payload),
+            area_uuid=self.device.uuid)
+
+    def _unregister(self, payload: Dict) -> None:
+        """Callback for the unregister redis command."""
+        self._is_registered = ExternalStrategyConnectionManager.unregister(
+            self.redis, self.channel_prefix, self.connected,
+            self._get_transaction_id(payload))
+
+    def _update_connection_status(self) -> None:
+        """Update the connected flag to sync it with the _registered flag.
 
-    def _unregister(self, payload):
-        self._connected = unregister_area(self.redis, self.channel_prefix, self.connected,
-                                          self._get_transaction_id(payload))
+        If the client unregistered (_registered=False), pop the asset from the aggregator mapping
```

Comment:
Also the other direction applies right? That is why you renamed the function, right?
So I would propose:
```suggestion
        Change assets' connection status including the connection to the aggregator.
```

Reply:
Right, Adapted!